ride/manager_vault.ride

```diff
@@ -0,0 +1,243 @@
+  throw(contractFilename + ": " + s)
+}
+
+# Manager Public Key
+func keyActiveManagerPublicKey()  = makeString(["%s", "managerPublicKey"], SEP)
+func keyPendingManagerPublicKey() = makeString(["%s", "pendingManagerPublicKey"], SEP)
+func keyCurrentManagerPublicKey() = makeString(["%s", "currentManagerPublicKey"], SEP)
+
+# Voting prefixes
+func keySetNewManagerVotePrefix(managerPublicKey: String) = makeString(["%s%s%s", "setManager", managerPublicKey], SEP)
+func keyDeactivateManagerVotePrefix()                     = makeString(["%s%s", "deactivateManager"], SEP)
+func keyActivateManagerVotePrefix()                       = makeString(["%s%s", "activateManager"], SEP)
+func keyAddNewAdminVotePrefix(adminCandidate: String)     = makeString(["%s%s%s", "addAdmin", adminCandidate], SEP)
+func keyRemoveAdminVotePrefix(removeCandidate: String)    = makeString(["%s%s%s", "removeAdmin", removeCandidate], SEP)
+func keyAllowedTxIdVotePrefix(txId: String)               = makeString(["%s%s%s", "allowTxId", txId], SEP)
+
+# Admin List key
+func keyAdminAddressList() = makeString(["%s", "adminAddressList"], SEP)
+
+# Allowed TXID key
+func keyAllowedTxId() = makeString(["%s", "TXID"], SEP)
+
+func getAdminVote(prefix: String, admin: String) = {
+  let voteKey = makeString([prefix, admin], SEP)
+  getInteger(voteKey).valueOrElse(0)
+}
+
+func pendingManagerPublicKeyOrUnit() = match keyPendingManagerPublicKey().getString() {
+  case s: String => s.fromBase58String()
+  case _: Unit => unit
+}
+
+func getAdminsList() = {
+  this.getString(keyAdminAddressList()).valueOrElse("").split(SEP)
+}
+
+func isInAdminList(address: String) = {
+  match (getAdminsList().indexOf(address)) {
+    case t:Int => true
+    case _ => false
+  }
+}
+
+# Generate List of keys with same prefix for all admins
+func genVotesKeysHelper(a: (List[String], String), adminAddress: String) = {
+  let (result, prefix) = a
+  (result :+ makeString([prefix, adminAddress], SEP), prefix)
+}
+func genVotesKeys(keyPrefix: String) = {
+  let adminList = keyAdminAddressList()
+  FOLD<20>(getAdminsList(), ([], keyPrefix), genVotesKeysHelper)
+}
+
+# Count all votes for Prefix
+func countVotesHelper(result: Int, voteKey: String) = {
+  result + getInteger(voteKey).valueOrElse(0)
+}
+func countVotes(prefix: String) = {
+  let votes = genVotesKeys(prefix)._1
+  FOLD<20>(votes, 0, countVotesHelper)
+}
+
+# Generate DeleteEntry for all votes with Prefix
+func clearVotesHelper(result: List[DeleteEntry], key: String) = {
+  result :+ DeleteEntry(key) 
+}
+func getClearVoteEntries(prefix: String) = {
+  let votes = genVotesKeys(prefix)._1
+  FOLD<20>(votes, [], clearVotesHelper)
+}
+
+func voteINTERNAL(
+    callerAddressString: String, 
+    keyPrefix: String, 
+    minVotes: Int, 
+    voteResult: 
+    List[StringEntry|IntegerEntry|DeleteEntry]
+  ) = {
+  let voteKey = makeString([keyPrefix, callerAddressString], SEP)
+  let adminCurrentVote = getAdminVote(keyPrefix, callerAddressString)
```

Comment:
if the getAdminVote function is used in one place, as it is now, you can make the response boolean. If the key value exists, return true